packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -971,11 +971,33 @@ abstract class PopupRoute<T> extends ModalRoute<T> {
 /// A [Navigator] observer that notifies [RouteAware]s of changes to the
 /// state of their [Route].
 ///
+/// [RouteObserver] informs subscribers whenever a route of type `T` is pushed
+/// on top of their own route of type `T` or popped from it. This is for example
+/// useful to keep track of page transitions, e.i. a `RouteObserver<PageRoute>`
+/// will inform subscribed [RouteAware]s whenever the user navigates away from
+/// the current page route to another page route.
```

Comment:
This seems like a weird feature. What if they're going from a PageRoute to a DialogRoute to a PageRoute to a PageRoute? Why should they only be notified once?